examples/source-to-image/assemble

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+set -x
+
+set -eo pipefail
+
+# Remove any 'environment.yml' or 'requirements.txt' files which may
+# have been carried over from the base image so we don't reinstall
+# packages which have already been installed. This could occur where
+# an S2I build was used to create a new base image with pre-installed
+# Python packages, with the new image then subsequently being used as a
+# S2I builder base image.
+
+rm -f /home/$NB_USER/work/environment.yml
+rm -f /home/$NB_USER/work/requirements.txt
+
+# Copy injected files to correct place in 'work' directory.
+
+cp -Rf /tmp/src/. /home/$NB_USER/work
+
+rm -rf /tmp/src
+
+# Install any Python modules. If we find an 'environment.yml' file we
+# assume we should use 'conda' to install packages. If 'requirements.txt'
+# use 'pip' instead. Ensure we are in the 'work' directory so relative
+# directory paths in these files resolve okay if installing packages
+# from subdirectories.
+
+if [ -f /home/$NB_USER/work/environment.yml ]; then
+    (cd /home/$NB_USER/work && conda env update --name root --file environment.yml)
```

Comment:
without spawning a shell, this can `conda env update -f /path/to/environment.yml`. Same below for pip.

Reply:
So changing sub directory in sub shell may not be required, never actually checked whether does the right thing and to be safe would always ensure in the same directory as source code and that file. If don't do things in ``work`` directory but ``$HOME`` then becomes moot as ``$HOME`` will be the current working directory anyway.

The ``--name root`` option is used because I have found that if don't have that and ``environment.yml`` has ``name`` attribute for environment, then it creates a separate environment under ``/opt/conda/envs``. I assumed this would not be want you want, because that environment isn't going to be activated and things would run in the root environment still, meaning installed packages would not be found. Thus to be safe was overriding things so always installed into the root environment.

Reply:
Yes, `--name` is required. Sorry, I didn't mean that my line was complete, just that you could replace cd in a sub-shell with an absolute path in the file argument. The full line:

    conda env update -n root -f /path/to/environment.yml

will work no matter what the cwd.